enhancements/installer/azure-cluster-scaling-for-outbound-traffic.md

```diff
@@ -0,0 +1,214 @@
+3. Installer sets up three additional public IP addresses for the cluster at install time. (Avoids scanerio 2 below)
+4. OpenShift install continues as normal.
+
+#### Variation [optional]
+
+A variation is to have a more opininated approach in the installer about the disableOutboundSNAT
+property instead of exposing it. In this approach would set the disableOutboundSNAT property based off
+other inputs in the install config.
+
+There are three scenarios in that the installer team should think about to create a more opinionated approach.
+
+Scenario 1 (Private cluster, LB outbound type):
+- There is an outbound rule on the LB
+- If you set `disableOutboundSNAT = true` in the cloud provider config and all public IP addresses 
+the installer creates via terraform, the explicit outbound rule will be used for egress
+
+Scenario 2 (public cluster, LB outbound type):
+- There is no outbound rule on the LB
+- If you set `disableOutboundSNAT = true` in the cloud provider config and all public IP addresses 
+the installer create via terraform set this to true as well, **egress will not work.**
+
+Scenario 3 (public cluster, LB outbound type):
+- There is no outbound rule on the LB
+- If you set `diableOutboundSNAT = true` **ONLY in the cloud provider config**, egress will work 
+through the terraform-created public IP addresses and egress will work.
+
+
+### API Extensions
+
+This enhancment would modify the [CloudProviderConfig API Object](https://github.com/openshift/installer/blob/master/pkg/asset/manifests/azure/cloudproviderconfig.go) to include the property `disableOutboundSNAT` in the config.
+
+```go
+
+config := config{
+		authConfig: authConfig{
+			Cloud:                       params.CloudName.Name(),
+			TenantID:                    params.TenantID,
+			SubscriptionID:              params.SubscriptionID,
+			UseManagedIdentityExtension: true,
+			UserAssignedIdentityID: "",
+		},
+		ResourceGroup:     params.ResourceGroupName,
+		Location:          params.GroupLocation,
+		SubnetName:        params.SubnetName,
+		SecurityGroupName: params.NetworkSecurityGroupName,
+		VnetName:          params.VirtualNetworkName,
+		VnetResourceGroup: params.NetworkResourceGroupName,
+		RouteTableName:    params.ResourcePrefix + "-node-routetable",
+		CloudProviderRateLimit:       false,
+		CloudProviderBackoff:         true,
+		CloudProviderBackoffDuration: 6,
+
+		UseInstanceMetadata: true,
+		LoadBalancerSku:             "standard",
+		ExcludeMasterFromStandardLB: &excludeMasterFromStandardLB,
+    DisableOutboundSNAT: true, // <-- Add this property
+	}
+
+
+```
+
+### Implementation Details/Notes/Constraints [optional]
+
+### Risks and Mitigations
+
+There is a risk that a customer with a public that the customer could set disableOutboundSNAT 
+to true, without having a public IP address to use. We propose having the installer create 
+IP addresses through terraform to mitgate this risk.
+
+### Drawbacks
```

Comment:
What is the perceived difference to traffic when disabling outbound SNAT? You talk about the benefits of disabling it, in that you can get extra IP addresses, but you don't talk about what we lose by disabling it?

If it's the recommended default by Microsoft, why are we going against their recommendation? I think some detail on the behavioural differences will help to clarify that